test/model.test.js

```diff
@@ -7557,6 +7557,20 @@ describe('Model', function() {
         assert.equal(usersFromDatabase[1].name, 'name from pre-save');
       });
     });
+    it('works if some document is not modified (gh-10437)', () => {
+      const userSchema = new Schema({
+        name: String
+      });
+
+      const User = db.model('User', userSchema);
+
+      return co(function*() {
+        const user = yield User.create({ name: 'Hafez' });
+
+        const res = yield User.bulkSave([user]);
+        assert.ok(res.result.ok);
```

Comment:
I'd prefer if we didn't assert on particular properties in `res.result`. I'd just make this check that no error was thrown using `const err = yield User.bulkSave([user]).then(() => null, err => err); assert.ifError(err);`

Reply:
I agree, that's the way to go.

Reply:
I agree, that's the way to go.